### PR TITLE
quickwit: fix compilation with rust 1.80

### PR DESCRIPTION
**Pull request by zimward**
## Description of changes
`Cargo.lock` has been patched to update the `time` crate and its dependencies to version `0.3.36` to fix the failing build with rust 1.80 (#332957).

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/servers/search/quickwit/update-time-for-rust-1.80.patch
Comment (SuperSandro2000) [inline pkgs/servers/search/quickwit/update-time-for-rust-1.80.patch:1]: I don't think we need the patch here, we can just update the lock file directly

Comment (zimward) [reply pkgs/servers/search/quickwit/update-time-for-rust-1.80.patch:1]: i tried to do so first but it then failed stating something along the lines of "source Cargo.lock is different from provided"

Comment (zimward) [reply pkgs/servers/search/quickwit/update-time-for-rust-1.80.patch:1]: Here is the exact error message:
```
Validating consistency between /build/source/quickwit/Cargo.lock and /build/cargo-vendor-dir/Cargo.lock
8228c8228
< version = "0.3.34"
---
> version = "0.3.36"
8230c8230
< checksum = "c8248b6521bb14bc45b4067159b9b6ad792e2d6d754d6c41fb50e29fefe38749"
---
> checksum = "5dfd88e563464686c916c7e46e623e520ddc6d79fa6641390f2e3fa86e83e885"
8261c8261
< version = "0.2.17"
---
> version = "0.2.18"
8263c8263
< checksum = "7ba3a3ef41e6672a2f0f001392bb5dcd3ff0a9992d618ca761a11c3121547774"
---
> checksum = "3f252a68540fde3a3877aeea552b832b40ab9a69e318efd078774a01ddee1ccf"

ERROR: cargoHash or cargoSha256 is out of date

Cargo.lock is not the same in /build/cargo-vendor-dir

To fix the issue:
1. Set cargoHash/cargoSha256 to an empty string: `cargoHash = "";`
2. Build the derivation and wait for it to fail with a hash mismatch
3. Copy the "got: sha256-..." value back into the cargoHash field
   You should have: cargoHash = "sha256-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX=";
```

Comment (SuperSandro2000) [reply pkgs/servers/search/quickwit/update-time-for-rust-1.80.patch:1]: 🤔 I think I had a similar problem when fixing wezterm. I think it got suggested to me to copy the already vendored Cargo.lock in postPatch to the normal location.

Comment (zimward) [reply pkgs/servers/search/quickwit/update-time-for-rust-1.80.patch:1]: I will look into it. Thanks for the pointer.

**Pull request merged**